worker/task/process_event_delivery.go

```diff
@@ -300,3 +280,65 @@ func parseAttemptFromResponse(m *datastore.EventDelivery, e *datastore.Endpoint,
 		UpdatedAt: primitive.NewDateTimeFromTime(time.Now()),
 	}
 }
+
+type EventDeliveryConfig struct {
+	group        *datastore.Group
+	subscription *datastore.Subscription
+}
+
+type RetryConfig struct {
+	Type       datastore.StrategyProvider
+	Duration   uint64
+	RetryCount uint64
+}
+
+type RateLimitConfig struct {
+	Count    int
+	Duration uint64
+}
+
+func (ec *EventDeliveryConfig) disableEndpoint() bool {
+	if ec.subscription.DisableEndpoint != nil {
+		return *ec.subscription.DisableEndpoint
+	}
+
+	return ec.group.Config.DisableEndpoint
+}
+
+func (ec *EventDeliveryConfig) retryConfig() (*RetryConfig, error) {
+	rc := &RetryConfig{}
+
+	if ec.subscription.RetryConfig != nil {
+		rc.Duration = ec.subscription.RetryConfig.Duration
+		rc.RetryCount = ec.subscription.RetryConfig.RetryCount
+		rc.Type = ec.subscription.RetryConfig.Type
+
+	} else if ec.group.Config != nil {
```

Comment:
@Dotunj Group Config should never be `nil`. At the point of creation, if the user does not set it, we should apply the default to the project. 

It should be an error scenario that a group has no configuration. It's similar to #956